src/neo-vm/Unsafe.cs

```diff
@@ -0,0 +1,78 @@
+﻿using System;
+using System.Runtime.CompilerServices;
+
+namespace Neo.VM
+{
+    internal static class Unsafe
+    {
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        unsafe public static bool Equals(byte[] x, byte[] y)
+        {
+            if (ReferenceEquals(x, y)) return true;
+            if (x is null || y is null) return false;
+            int len = x.Length;
+            if (len != y.Length) return false;
+            fixed (byte* xp = x, yp = y)
+            {
+                long* xlp = (long*)xp, ylp = (long*)yp;
```

Comment:
We need to check if the length is more than 8 bytes, otherwise we are taking random memory

Reply:
https://github.com/neo-project/neo-vm/blob/81c3358a5b8c1611321527cd790465417e4a5cbc/src/neo-vm/Unsafe.cs#L18

This line has checked it already.